File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/pairakabe/domain/PairakabeObject.java
package com.zwstudio.logicpuzzlesandroid.puzzles.pairakabe.domain;

public abstract class PairakabeObject {
    public abstract String objAsString();
    public static PairakabeObject objTypeFromString(String str) {
        switch (str) {
            case "wall":
                return new PairakabeWallObject();
            case "marker":
                return new PairakabeMarkerObject();
            default:
                return new PairakabeEmptyObject();
        }
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/pairakabe/PairakabeObject.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.pairakabe

import com.zwstudio.logicpuzzlesandroid.common.domain.Position

enum class PairakabeObject {
    Empty, Hint, Marker, Wall
}

class PairakabeGameMove(val p: Position, var obj: PairakabeObject = PairakabeObject.Empty)
